bridges/centralized-ethereum/src/actors/wit_poller.rs

```diff
@@ -130,15 +133,58 @@ impl WitPoller {
                         );
 
                         let result = tally.tally;
-                        dr_reporter_addr
-                            .send(DrReporterMsg {
-                                dr_id,
-                                dr_bytes,
-                                dr_tx_hash,
-                                result,
-                            })
-                            .await
-                            .unwrap();
+                        // Get timestamp of first block with commits. The timestamp of the data
+                        // point is the timestamp of that block minus 45 seconds, because the commit
+                        // transactions are created one epoch earlier.
+                        // TODO: first block with commits is hard to obtain, we are simply using the
+                        // block that included the data request.
+                        let timestamp = {
+                            let method = String::from("getBlock");
+                            let params = json!([dr_block_hash]);
+                            let req = jsonrpc::Request::method(method)
+                                .timeout(Duration::from_millis(5_000))
+                                .params(params)
+                                .expect("params failed serialization");
+                            let report = witnet_client.send(req).await;
+                            let report = match report {
+                                Ok(report) => report,
+                                Err(_) => {
+                                    log::error!(
+                                        "Failed to connect to witnet client, will retry later"
+                                    );
+                                    break;
+                                }
+                            };
+                            let block = match report {
+                                Ok(value) => serde_json::from_value::<Block>(value)
+                                    .expect("failed to deserialize block"),
+                                Err(e) => {
+                                    log::error!(
+                                        "error in getBlock call ({}): {:?}",
+                                        dr_block_hash,
+                                        e
+                                    );
+                                    continue;
+                                }
+                            };
+
+                            let block_number = block.block_header.beacon.checkpoint;
+                            // TODO: get constants from somewhere instead of hardcoding them here?
```

Comment:
![image](https://user-images.githubusercontent.com/44392885/159505666-0ae44055-11fb-48d1-9a99-17ef9d80ff96.png)
There is a jsonrpc method that returns the consensus constants